src/scipp/core/deprecation.py

```diff
@@ -0,0 +1,49 @@
+import warnings
+
+from .util import VisibleDeprecationWarning
+
+
+def _warn_attr_removal():
+    warnings.warn(
+        "sc.DataArray.attrs has been deprecated and will be removed in Scipp v24.12.0. "
```

Comment:
Do we need to make the period this long? I didn't think we would go longer than 1y.

Reply:
By the time we released this... and 2023 is almost over anyway. Do you see any harm in having it this long?

Reply:
Not directly. It's just nice to get these things out of the way to simplify the code.